community/bolt/src/main/java/org/neo4j/bolt/v1/runtime/TransactionStateMachine.java

```diff
@@ -95,6 +95,14 @@ public void streamResult( ThrowingConsumer<BoltResult, Exception> resultConsumer
         }
     }
 
+    /**
+     * Rollback and close transaction. Move back to {@link State#AUTO_COMMIT}.
+     * <p>
+     * <b>Warning:</b>This method should only be called by the bolt worker thread during it's regular message
+     * processing. It is wrong to call it from a different thread because kernel transactions are not thread-safe.
```

Comment:
I think by doing that we will guard us against misuse since the comment is something that is not enforced; for sure that can be separate PR.